torchaudio/transforms.py

```diff
@@ -567,6 +569,61 @@ def forward(self, x_mu: Tensor) -> Tensor:
         """
         return F.mu_law_decoding(x_mu, self.quantization_channels)
 
+class ALawEncoding(torch.nn.Module):
+    r"""Encode signal based on A-law companding.  For more info see the
+    `Wikipedia Entry <https://en.wikipedia.org/wiki/A-law_algorithm>`_
+
+    This algorithm assumes the signal has been scaled to between -1 and 1 and
+    returns a signal encoded with values from 0 to quantization_channels - 1
+
+    Args:
+        quantization_channels (int, optional): Number of channels. (Default: ``256``)
+    """
+    __constants__ = ['quantization_channels']
+
+    def __init__(self, quantization_channels: int = 256, compress_param: float = 83.7) -> None:
+        super(ALawEncoding, self).__init__()
+        self.quantization_channels = quantization_channels
+        self.compress_param = compress_param
+
+    def forward(self, x: Tensor) -> Tensor:
+        r"""
+        Args:
+            x (Tensor): A signal to be encoded.
+
+        Returns:
+            x_a (Tensor): An encoded signal.
+        """
+        return F.a_law_encoding(x, self.quantization_channels, self.compress_param)
+
+
+class ALawDecoding(torch.nn.Module):
+    r"""Decode A-law encoded signal.  For more info see the
+    `Wikipedia Entry <https://en.wikipedia.org/wiki/A-law_algorithm>`_
+
+    This expects an input with values between 0 and quantization_channels - 1
+    and returns a signal scaled between -1 and 1.
+
+    Args:
+        quantization_channels (int, optional): Number of channels. (Default: ``256``)
+    """
+    __constants__ = ['quantization_channels']
+
+    def __init__(self, quantization_channels: int = 256, compression_param: float = 83.7) -> None:
+        super(ALawDecoding, self).__init__()
+        self.quantization_channels = quantization_channels
+        self.compression_param = compression_param
+
+    def forward(self, x_mu: Tensor) -> Tensor:
+        r"""
+        Args:
+            x_a (Tensor): An A-law encoded signal which needs to be decoded.
+
+        Returns:
+            Tensor: The signal decoded.
+        """
+        return F.a_law_decoding(x_a, self.quantization_channels, self.compression_param)
+
```

Comment:
Is `x_mu` equal to `x_a`? How about just using `x`? this seems to lead to [error](https://app.circleci.com/pipelines/github/pytorch/audio/3546/workflows/7f02a96c-2ad6-426c-b809-41fc6e12c192/jobs/104322)

Reply:
I named the input `x_a` since it is an `x` encoded with `a` law. In `MuLaw` it is called `x_mu` so I wanted to name it consistently